### PR TITLE
Fixing the bug where nested describe namespace was flattened.

### PR DESCRIPTION
**Pull request by jpaulin**
This code handles both degenerate (zero-length) and normal cases
of array of test suite names.

Somewhere from 0.3.3 onwards there was a bug creeping in. This caused the 
namespace to flatten, ie. only the first word of a namespace was included. This corresponds
to a flat namespace. 

Nested describe's (per Jasmine language) were being flattened thus and overlaps of names
started happening.

### CODE REVIEW

**Comment by jpaulin** [comment]
CLA signed.

**Comment by Dignifiedquire** [comment]
Thanks, will review doon. But why did you close the PR?

**Comment by jpaulin** [comment]
Sorry, I thought this was part of the workflow. :) 
What I did was try to now put a new pull request to fix my typos per Javascript (I used semicolons etc.)
Can you see the latest, as passing the automation on syntax?

**Comment by jpaulin** [comment]
[Clicked] It's open. :) As said, this is very educational. Getting the hang of it.

**Comment by jpaulin** [comment]
Hygienics done! 

Really excited to see what comes. I'll be here all night along.

File: index.js [hunk 1]
```diff
@@ -83,8 +83,9 @@ var JUnitReporter = function (baseReporterDecorator, config, logger, helper, for
 
   var getClassName = function (browser, result) {
     var browserName = browser.name.replace(/ /g, '_').replace(/\./g, '_') + '.'
-
-    return (useBrowserName ? browserName : '') + (pkgName ? pkgName + '.' : '') + result.suite[0]
+    var namesNested = ''
+    if (result.suite && result.suite.length > 0) { namesNested = result.suite.join(' ') }
+    return (useBrowserName ? browserName : '') + (pkgName ? pkgName + '.' : '') + namesNested
```
Comment (Dignifiedquire) [inline index.js]: Could you clean this code up a bit, while we are at it? Something like this

```js
var name = ''

if (useBrowserName) {
  name += browserName
}

if (pkgName) {
  name += '.'
}

if (result.suite && result.suite.length > 0) {
  name += result.suite.join(' ')
}

return name
```

**Comment by jpaulin** [comment]
@Dignifiedquire Done refactoring.

**Comment by jpaulin** [comment]
If you want any comments etc feel free. I'm at it. :)

**Comment by Dignifiedquire** [comment]
Sorry for not responding earlier. This looks good to me. You mentioned something about testing this, that would be nice but I'm not sure how to do that as I can't reproduce the error myself at the moment.

**Comment by jpaulin** [comment]
It would require having the reporter be run on three cases I think:
- flat describe structure (just plain Jasmine, with no nesting, just a flat describe() and with it normal it() specs. 
- 2. level nested describe (having one level of nesting)
- 3. level nested describe (kind of 'checking the iteration beyond obvious') 
Where could I try these?

**Comment by dignifiedquire** [comment]
@jpaulin I would suggest adding those into this repo, so we always run them, it's okay if the we have to manually review the output for now.

**Comment by jpaulin** [comment]
I'll start doing the change now. Thanks for getting back on this. 

I'll check the Travis part more, since that's kind of new for me. Travis is used automatically in all Github projects as the CI engine. I generally know CI systems quite well, but it's the particulars that make me first read the introduction.

Btw, any chance I can live with current devDependencies and **not **add any new?

**Comment by dignifiedquire** [comment]
> I'll check the Travis part more, since that's kind of new for me. Travis is used automatically in all Github projects as the CI engine. I generally know CI systems quite well, but it's the particulars that make me first read the introduction.

It's pretty straightforward it'll just give a linux env executing the commands you give it. Also it gives a Firefox to run karma in a real browser which is nice. Checkout out this [`.travis.yml`](https://github.com/karma-runner/karma-mocha/blob/master/.travis.yml) from karma-mocha for some details. Also the travis docs are pretty decent: https://docs.travis-ci.com/user/getting-started/

> Btw, any chance I can live with current devDependencies and *not *add any new?

Not sure what you mean with that

**Comment by jpaulin** [comment]
> Not sure what you mean with that

I'm trying to design a test that fails in case this PR code wouldn't solve what it was meant for. So the initial idea is to have at least 2 Jasmine flavor test file (first: "depth one", ie. use of 'describe' inside another 'describe') and the second for depth two. If the new code now under work fails to produce output, the output would either be ~ non-existent, or still be flattened. That would be the contents of the test.

The question I have is that does Travis basically contain already all that's required?

**Comment by dignifiedquire** [comment]
Just to clarify travis just runs what we have in this repo. At the moment there is no real test setup in this repo, aka there is no karma started with the reporter and executed. So for this you would have to set this up, you can take a look karma-mocha for an example how this is done. So the idea is that this test runs on your local machine on calling `grunt` like it does linting at the moment. And if that works they should work on travis as well.

**Comment by jpaulin** [comment]
Very good tip, and probably just saved tons of my forehead bumping to wall! I would've been looking for that ghost runner for a looong time. Now will go for local first.

**Comment by jpaulin** [comment]
I think this PR can be closed, superseded by recent activity in PR #85

**Comment by dignifiedquire** [comment]
#85 was merged so closing this one.

**Pull request closed without merging**